models/pull_sign.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+)
+
+// SignMerge determines if we should sign a PR merge commit to the base repository
+func (pr *PullRequest) SignMerge(u *User, tmpBasePath, baseCommit, headCommit string) (bool, string) {
```

Comment:
How about move these new codes to `services/pull`?

Reply:
So SignInitialCommit has to currently be in models and given the rest of this code is in models - it probably makes more sense to keep it there until we can get SignInitialCommit out.

Reply:
I think we can move them step by step so that it will be easy to review. There is some intermediate state that some parts in models, some parts in pull but I think that's OK.